app/src/main/java/com/github/se/eduverse/viewmodel/ProfileViewModel.kt

```diff
@@ -279,6 +285,61 @@ open class ProfileViewModel(private val repository: ProfileRepository) : ViewMod
   open fun resetDeleteState() {
     _deletePublicationState.value = DeletePublicationState.Idle
   }
+
+  /**
+   * Loads all publications that a user has favorited. Combines favorite IDs with full publication
+   * details.
+   *
+   * @param userId The ID of the user whose favorites should be loaded
+   */
+  fun loadFavoritePublications(userId: String) {
+    viewModelScope.launch {
+      try {
+        // Get IDs of favorited publications
+        val favoriteIds = repository.getFavoritePublicationsIds(userId)
+        // Get all publications to find the full details
+        val allPublications = repository.getAllPublications()
+        // Filter to only favorited publications
+        val favoritePublicationsList = allPublications.filter { it.id in favoriteIds }
```

Comment:
Instead of loading every publications and filtring to only take the favorites, you could define a new method and filter directly from the querry (`publicationCollection.whereIn("id", favoriteIds).get()...` to be more efficent

Reply:
Ok im gonna try to do this